### ISSUE TITLE
twitter.com takes forever to load

### ISSUE THREAD

**Issue by Gozala** [servo/servo#11592]
Loading https://twitter.com/ in servo takes forever, while in all other browsers it's almost spontaneous. Unlike #11223 it does finish load though. I also observe following errors in the output which may provide some clues to a problem:

```
ERROR:js::rust: Error at https://twitter.com/:2:187: invalid range in character class
ERROR:js::rust: Error at https://abs.twimg.com/c/swift/en/init.cb652131421cd60235d870458173a76067f43b9b.js:430:109: too much recursion
```

P.S.: loading other links like https://twitter.com/rustlang does not seem to have same problem, so depending where you at you may or may not be able to reproduce

**Comment by jdm** [comment]
Are you referring to the homepage, or the logged-in view?

**Comment by Gozala** [comment]
Home page

Typed on tiny keyboard

**Comment by Coder206** [comment]
@Gozala What OS are you using to twitter.com?

**Comment by Coder206** [comment]
@jdm Do you know if Servo supports and unlimited number of classes?

**Comment by jdm** [comment]
@Coder206 I'm not sure what you're asking. Could you rephrase the question?

**Comment by Coder206** [comment]
@jdm For the OS thing or CSS?

**Comment by Coder206** [comment]
@jdm @Gozala 
OS: @Gozala Which OS are you using when trying to get https://twitter.com?
CSS: I noticed there are two CSS classes that are being expressed on the Twitter page, but they are they have many characters. (from the webpage: class="typeahead-item typeahead-dm-conversation-item") So my question was is there a limit on CSS classes like a character length or number of classes?

**Comment by metajack** [comment]
@Coder206 @SimonSapin would be able to answer your CSS question. Regardless of whether there is a limit in the spec and implementation, we still have to parse the thing. I suppose you could make one that is so large to run the parser out of memory. I don't think that would be causing the page to close down.

Probably the first thing to be done is view the profiler output in the HTML tool to see what's actually going on. Previously we've seen too many resource requests and other such behavior as culprits.

**Comment by Coder206** [comment]
@metajack That makes sense!

**Comment by Gozala** [comment]
> @Gozala What OS are you using to twitter.com?

I'm on OSX

**Comment by Coder206** [comment]
@Gozala Here is the view on my system, a VM running Fedora. Please note I am also receiving the errors you described above.

![servotwitter](https://cloud.githubusercontent.com/assets/7758155/15795346/bc3ee68a-29be-11e6-9e85-4e26ed5f6c44.png)

**Comment by Gozala** [comment]
@Coder206 I get these number son my machine:

```
time servo -w https://twitter.com
ERROR:js::rust: Error at https://twitter.com/:2:187: invalid range in character class

ERROR:js::rust: Error at https://abs.twimg.com/c/swift/en/init.fabcd7d9ac71b5a695afe033d0f3de85b2d4088e.js:430:109: too much recursion

servo -w https://twitter.com  12.52s user 1.70s system 117% cpu 12.058 total
```

On safari it usually loads with in 2 to 3 seconds. I think that too much recursion is probably what makes it feel much worse as it takes most of my CPU and machine becomes unresponsive so 12 secs feel a lot longer.

**Comment by Coder206** [comment]
@Gozala I know what you mean! I am getting feeling a similar duration on Windows 10.

**Comment by dd0x68** [comment]
@metajack @Gozala I have a profiler trace [here](https://www.dropbox.com/s/y6hcer2ijjbxsdi/twitter.html?dl=0) on OSX

Total time: 12.104185 s

The main problem seems to be that the operations are very serial. There are also some single operations that are taking a lot of time:
-LayoutPerform task 1891.074787 ms
-ScriptParseHTML task 1891.635303 ms
-NetHTTPRequestResponse: 698.675547 ms (isn't the longest, but seems to be a blocker)

The initial request takes 829.73298 ms before anything can happens.

I'm looking into some of the network performance issues, and this is one of my main test examples.

**Comment by jdm** [comment]
Curious - every attempt at reproducing this so far has given me a fully loaded page in about 1s in a release build on master.

**Comment by Coder206** [comment]
@Gozala Did you try re cloning the repository and try building and running again?

**Comment by nox** [comment]
Instantaneous for me now.
